### ISSUE TITLE
Failed account creation

### ISSUE THREAD

**Issue by rhomsher** [USEPA/GeoPlatform-Request-Tool#182]
In recent tests in GeoSecure when approving a request for an existing account, the approval fails with an unspecified error. It may be an issue specifically with adding users to Shared Update groups. A guess is that it's an order of operations issue where the user's role needs to be updated to GeoSecure Data Editor first, and then added to the corresponding Shared Update group (which can only receive members with a Creator user type).

**Comment by tbock** [comment]
Yes but the user type has to be changed to one compatible with the role first.  This is one of the reasons the issue is an enhancement as the user type has never been a configuration point in this tool.  It is in the data model but has never been offered as a configuration option in the interface.  The second issue is the hierarchy of the roles themselves.  Currently the role is only used for the original account creation not for assigning existing users to existing groups.  If we are to enable this functionality of setting a users type and role when approving an existing account through the tool; I would presume we wouldn't want a users role to be demoted from editor to viewer if they where approved on a project that was configured with the viewer role. 

To enable this functionality we would need to:
- [ ] add user type as a configurable option similar to how roles work now.
- [ ] determine how best to check/capture and/or configure for role and type compatibility 
- [ ] add the ability to configure role hierarchy

Then if someone is approved for a response/project that has a role of a higher level we can check to see if they have a compatible user type, change the user type if necessary, and then change the role.  If the role for the project/response is lower we can ignore. 

Does this sound correct? @rhomsher
